pkg/virt-api/rest/subresource.go

```diff
@@ -239,26 +239,19 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 		return
 	}
 
-	startOnly := false
 	vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
 	if err != nil {
 		if !errors.IsNotFound(err) {
 			response.WriteError(http.StatusInternalServerError, err)
 			return
 		}
-		// If there's no VMI, just request to start
-		startOnly = true
+		response.WriteError(http.StatusBadRequest, fmt.Errorf("VM is not running"))
```

Comment:
could you update https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-api/api.go#L205 with `Returns(http.StatusBadRequest, "Bad request", nil)` (the same for start and stop requests) and run `make generate`?

Reply:
good catch :+1: